skyrim-platform-interfaces/skyrim-ae/src/Papyrus/Extensions/TESModPlatform.h

```diff
@@ -0,0 +1,756 @@
+} */
+
+/* uint32_t TESModPlatform::GetNthVtableElement(RE::BSScript::IVirtualMachine*
+vm, RE::VMStackID stackId, RE::StaticFunctionTag*, RE::TESForm* pointer,
+uint32_t pointerOffset, uint32_t elementIndex)
+{
+  static auto getNthVTableElement = [](void* obj, size_t idx) {
+    using VTable = size_t*;
+    auto vtable = *(VTable*)obj;
+    return vtable[idx];
+  };
+
+  if (pointer && elementIndex >= 0) {
+    __try {
+      return getNthVTableElement(reinterpret_cast<uint8_t*>(pointer) +
+                                   pointerOffset,
+                                 elementIndex) -
+        REL::Module::BaseAddr();
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
+    }
+  }
+  return -1;
+} */
+
+/* RE::Module::BaseAddr() doesnt exist anymore */
+uint32_t GetNthVtableElement(VM* vm, StackID stackId, RE::StaticFunctionTag*,
+                             RE::TESForm* pointer, uint32_t pointerOffset,
+                             uint32_t elementIndex)
+{
+  static auto getNthVTableElement = [](void* obj, size_t idx) {
+    using VTable = size_t*;
+    auto vtable = *(VTable*)obj;
+    return vtable[idx];
+  };
+
+  if (pointer && elementIndex >= 0) {
+    __try {
+      return getNthVTableElement(reinterpret_cast<uint8_t*>(pointer) +
+                                   pointerOffset,
+                                 elementIndex) -
+        REL::Module::get().base(); // ???
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
+    }
+  }
+  return -1;
+}
+
+/* bool TESModPlatform::IsPlayerRunningEnabled(RE::BSScript::IVirtualMachine*
+vm, RE::VMStackID stackId, RE::StaticFunctionTag*)
+{
+  if (auto controls = RE::PlayerControls::GetSingleton())
+    return controls->data.running;
+  return false;
+} */
+
+/* no changes */
+bool IsPlayerRunningEnabled(VM* vm, StackID stackId, RE::StaticFunctionTag*)
+{
+  if (auto controls = RE::PlayerControls::GetSingleton())
+    return controls->data.running;
+  return false;
+}
+
+/* RE::BGSColorForm* TESModPlatform::GetSkinColor(
+  RE::BSScript::IVirtualMachine* vm, RE::VMStackID stackId,
+  RE::StaticFunctionTag*, RE::TESNPC* base)
+{
+  auto factory = IFormFactory::GetFactoryForType(BGSColorForm::kTypeID);
+  if (!factory)
+    return nullptr;
+  auto col = (RE::BGSColorForm*)factory->Create();
+  if (!col)
+    return nullptr;
+  col->color = base->bodyTintColor;
+  return col;
+} */
+
+/* doesn't use SKSE now */
+/* streamlined function name with the rest of them */
+RE::BGSColorForm* TESModPlatform::GetNPCSkinColor(VM* vm, StackID stackId,
+                                                  RE::StaticFunctionTag*,
+                                                  RE::TESNPC* npc)
+{
+  auto cf = Utils::CreateForm<RE::BGSColorForm>(RE::FormType::ColorForm);
+  if (!cf)
+    return nullptr;
+  cf->color = npc->bodyTintColor;
+  return cf;
+}
+
+/* since this is papyrus RE::BGSColorForm* colorForm looks more logical... or
+ * not? */
+void SetNPCSkinColor(VM* vm, StackID stackId, RE::StaticFunctionTag*,
```

Comment:
Please be careful and do not introduce backward-incompatible changes.
Porting to AE is already a problem. Not sure we need to introduce compatibility problems for Papyrus functions.
I didn't see if more signatures have changed, but changing this signature is guaranteed to break skymp. 
So please consider keeping from changing the API since there is a chance that we mistakenly approve it on review.

Reply:
Why it wasn't color form initially? AFAIK we didn't want to mess with finding/allocation color forms. NPC color comes from the server in our case.